pkg/openai/main.go

```diff
@@ -87,15 +85,22 @@ func (c *Connector) CreateExecution(defUID uuid.UUID, task string, config *struc
 }
 
 // NewClient initializes a new OpenAI client
-func NewClient(apiKey, org string) Client {
+func NewClient(apiKey, org string, logger *zap.Logger) Client {
```

Comment:
`logger` is passed here as an argument, as it's the strategy in this repo. In e.g. `pipeline-backend` we use `zap.GetLogger`, which would be possible here. Would it make sense to unify the strategies?

Reply:
Yes, we need to propagate the logger from pipeline-backend to here.

Reply:
👍 There's [a ticket](https://linear.app/instill-ai/issue/INS-2960/propagate-logger-to-connectors-potentially-to-any-component) for this, it can be tackled separately or as the starting point of the next error handling PR (my plan was validating and merging this, then working a new PR with one commit per connector)

Reply:
Great!